app/src/main/java/com/example/mytrainingpal/components/GifComponent.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.mytrainingpal.components
+
+import android.os.Build
+import androidx.compose.foundation.Image
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import coil.ImageLoader
+import coil.compose.rememberAsyncImagePainter
+import coil.decode.GifDecoder
+import coil.decode.ImageDecoderDecoder
+import coil.request.ImageRequest
+
+// Reference: Dr BigKitkat answer on github + Coil library docs
+// https://stackoverflow.com/questions/60229555/adding-gif-into-jetpack-compose
+// about AsyncImagePainter --> https://coil-kt.github.io/coil/compose/
+
+
+@Composable
+fun GifImage(
+    size: Int,
+    // TODO: gifs saved under R.drawable.<gifName> (added through the GUI Resource Manager)
+    gifPath: Any?
```

Comment:
I would change the order of the arguments to make the most significant argument (the gif path) to be first. Note that I also made the path a string. 

```suggestion
    gifPath: String,
    size: Int
```